Framework/Reflectometry/src/ReflectometryWorkflowBase2.cpp

```diff
@@ -83,10 +83,51 @@ std::vector<size_t> getProcessingInstructionsAsIndices(std::string const &instru
   });
   return indices;
 }
+
+/** Get a detector parameter from an instrument
+ *
+ * @param instrument : the instrument containing the parameters
+ * @param param : the parameter name
+ * @return : the parameter value converted to an integer, or boost::none if it was not found
+ * @throw : if the detector index is invalid
+ */
+boost::optional<size_t> getDetectorParamOrNone(const Instrument_const_sptr &instrument,
+                                               const MatrixWorkspace_sptr &inputWS, const std::string &param) {
+  const std::vector<double> value = instrument->getNumberParameter(param);
+  if (value.empty()) {
+    return boost::none;
+  }
+  // Check it's a valid workspace index
+  if (value[0] < 0) {
+    std::ostringstream msg;
+    msg << "Parameter file value " << param << "=" << value[0] << " is invalid; it must be 0 or greater";
+    throw std::out_of_range(msg.str());
+  }
+
+  auto const wsIndex = static_cast<size_t>(value[0]);
+  if (wsIndex >= inputWS->getNumberHistograms()) {
+    std::ostringstream msg;
+    msg << "Parameter file value " << param << "=" << wsIndex
+        << " is out of range; max workspace index= " << inputWS->getNumberHistograms() - 1 << ")";
+    throw std::out_of_range(msg.str());
```

Comment:
If anything it's probably less efficient but won't make much difference in this case. I use it just because it automatically converts the numbers so avoids having to do the `std::to_string` calls. I just think it makes it a bit more readable when you've got a lot of arguments.

Reply:
Ah ok, that makes sense - thanks!